Code/Framework/AzCore/AzCore/Statistics/StatisticalProfilerProxy.h

```diff
@@ -135,30 +131,55 @@ namespace AZ::Statistics
         StatisticalProfilerProxy(StatisticalProfilerProxy&&) = delete;
         StatisticalProfilerProxy& operator=(StatisticalProfilerProxy&&) = delete;
 
+        void RegisterProfilerId(StatisticalProfilerId id)
+        {
+            m_profilers.try_emplace(id, AZStd::move(ProfilerInfo()));
+        }
+
         bool IsProfilerActive(StatisticalProfilerId id) const
         {
-            return m_activeProfilersFlag[static_cast<AZStd::size_t>(id)];
+            auto iter = m_profilers.find(id);
+            return (iter != m_profilers.end()) ? iter->second.m_enabled : false;
         }
 
         StatisticalProfilerType& GetProfiler(StatisticalProfilerId id)
         {
-            return m_profilers[static_cast<AZStd::size_t>(id)];
+            auto iter = m_profilers.try_emplace(id, AZStd::move(ProfilerInfo())).first;
+            return iter->second.m_profiler;
         }
 
-        void ActivateProfiler(StatisticalProfilerId id, bool activate)
+        void ActivateProfiler(StatisticalProfilerId id, bool activate, bool autoCreate = true)
         {
-            m_activeProfilersFlag[static_cast<AZStd::size_t>(id)] = activate;
+            ProfilerMap::iterator iter;
+            if (autoCreate)
+            {
+                iter = m_profilers.try_emplace(id, AZStd::move(ProfilerInfo())).first;
+            }
+            else
+            {
+                iter = m_profilers.find(id);
+            }
+            iter->second.m_enabled = activate;
         }
 
         void PushSample(StatisticalProfilerId id, const StatIdType& statId, double value)
         {
-            m_profilers[static_cast<AZStd::size_t>(id)].PushSample(statId, value);
+            if (auto iter = m_profilers.find(id); iter != m_profilers.end())
+            {
+                iter->second.m_profiler.PushSample(statId, value);
+            }
         }
 
     private:
-        // TODO:BUDGETS the number of bits allocated here must be based on the number of budgets available at profiler registration time
-        AZStd::bitset<128> m_activeProfilersFlag;
-        AZStd::vector<StatisticalProfilerType> m_profilers;
+        struct ProfilerInfo
+        {
+            StatisticalProfilerType m_profiler;
+            bool m_enabled{ false };
+        };
+
+        using ProfilerMap = AZStd::unordered_map<StatisticalProfilerId, ProfilerInfo>;
```

Comment:
If profilers are infrequently added, using a `vector`, sorting the profilers before use and doing a binary search may work out to be more efficient. Definitely profile though.

Reply:
Profilers are added on-demand as Budgets are defined/used so that's something I'll look into.